src/Spatial/Euclidean/StraightLine2D.cs

```diff
@@ -0,0 +1,315 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace MathNet.Spatial.Euclidean
+{
+    /// <summary>
+    /// Represents a line conforming to the equation y = mx + b where m is the gradient and b is the y-intercept
+    /// </summary>
+    public struct StraightLine2D : ILine
```

Comment:
I think we want ` : IEquatable<StraightLine2D>` alo here to avoid boxing.

Reply:
ok